### PR TITLE
Fixes for omnisharp paths on Windows and support for omnisharp-dotnet

### PR DESCRIPTION
**Pull request by JordanAnthonyKing**
Fix install paths for omnisharp on windows. Add support for installing omnisharp-dotnet (6.0). And fix omnisharp exe path on windows.

### CODE REVIEW

File: langserver/omnisharp-dotnet_windows.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "omnisharp-dotnet",
+    "languageId": "c#",
+    "command": ["C:\\Users\\Jordan\\.emacs.d\\.cache\\omnisharp\\OmniSharp.exe",
```
Comment (manateelazycat) [inline langserver/omnisharp-dotnet_windows.json:4]: C:\\Users\\Jordan replace with `%USERPROFILE%` , will work?

File: langserver/omnisharp-mono_windows.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "omnisharp-mono",
+    "languageId": "c#",
+    "command": ["mono",
+                ".cache\\omnisharp\\OmniSharp.exe",
+                "-z",
+                "-lsp"],
+    "settings": {}
+}
```
Comment (manateelazycat) [inline langserver/omnisharp-mono_windows.json:9]: If mono can't work, we should remove omnisharp-mono-*.json.

File: langserver/omnisharp-dotnet.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "omnisharp-dotnet",
+    "languageId": "c#",
+    "command": ["$HOME/.emacs.d/.cache/omnisharp/OmniSharp.exe",
```
Comment (manateelazycat) [inline langserver/omnisharp-dotnet.json:4]: Use `%USER_EMACS_DIRECTORY%/.cache/omisharp/OmniSharp.exe` instead.

File: langserver/omnisharp-dotnet_windows.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "omnisharp-dotnet",
+    "languageId": "c#",
+    "command": [".cache\\omnisharp\\OmniSharp.exe",
```
Comment (manateelazycat) [inline langserver/omnisharp-dotnet_windows.json:4]: Use `%USER_EMACS_DIRECTORY%\\.cache\\omisharp\\OmniSharp.exe` instead.

File: lsp_bridge.py [hunk 5]
```diff
@@ -191,6 +191,11 @@ def _open_file(self, filepath):
                 return False
             
             lang_server_info = load_single_server_info(single_lang_server)
+            if "omnisharp" in single_lang_server and get_os_name() == "windows":
+                user_emacs_dir = get_emacs_func_result("get-user-emacs-directory")
+                full_path = str(pathlib.Path(user_emacs_dir, lang_server_info["command"][0]))
+                lang_server_info["command"][0] = full_path
+
```
Comment (manateelazycat) [inline lsp_bridge.py:198]: We replace `%USER_EMACS_DIRECTORY%` with get-user-emacs-directory here, not write `omnisharp` here, lsp-bridge should not hardcode lsp server name.

**Review by manateelazycat — CHANGES_REQUESTED** [review]

File: lsp_bridge.py [hunk 6]
```diff
@@ -191,6 +191,13 @@ def _open_file(self, filepath):
                 return False
             
             lang_server_info = load_single_server_info(single_lang_server)
+
+            windows_user_emacs_dir_keyword = "%USER_EMACS_DIRECTORY%"
+            if windows_user_emacs_dir_keyword in lang_server_info["command"][0]:
+                user_emacs_dir = get_emacs_func_result("get-user-emacs-directory").replace("/", "\\")
```
Comment (manateelazycat) [inline lsp_bridge.py:197]: Why need replace / with \\ ? 

\\ is window path splitter, It won't works with Linux.

**Review by manateelazycat — CHANGES_REQUESTED** [review]

**Comment by manateelazycat** [comment]
@JordanAnthonyKing If omnisharp can works with out of box and don't need mono, we need remove omnisharp-mono-*.json files from patch.

**Comment by JordanAnthonyKing** [comment]
> @JordanAnthonyKing If omnisharp can works with out of box and don't need mono, we need remove omnisharp-mono-*.json files from patch.

Mono is used for older, .net framework projects, as well as the unity game engine. It will be useful to some.

**Pull request merged**